src/libsync/abstractnetworkjob.cpp

```diff
@@ -164,6 +164,27 @@ void AbstractNetworkJob::slotFinished()
         qCWarning(lcNetworkJob) << "SslHandshakeFailedError: " << errorString() << " : can be caused by a webserver wanting SSL client certificates";
     }
 
+    // Qt doesn't yet transparently resend HTTP2 requests, do so here
+    QByteArray verb = requestVerb(*reply());
+    if (_reply->error() == QNetworkReply::ContentReSendError
+        && _reply->attribute(QNetworkRequest::HTTP2WasUsedAttribute).toBool()
+        && (!_requestBody || _requestBody->isSequential())
+        && !verb.isEmpty()) {
+        qCInfo(lcNetworkJob) << "HTTP2 resending" << _reply->request().url();
+        resetTimeout();
+        if (_requestBody) {
+            if(!_requestBody->isOpen())
+               _requestBody->open(QIODevice::ReadOnly);
+            _requestBody->seek(0);
+        }
+        sendRequest(
+            verb,
+            _reply->request().url(),
+            _reply->request(),
+            _requestBody);
+        return;
+    }
+
     if (_reply->error() != QNetworkReply::NoError) {
```

Comment:
The previous chunk of code could be moved in this block, maybe

Reply:
I don't want to do it because it'd increase nesting unnecessarily.